packages/fuel-indexer-api-server/src/uses.rs

```diff
@@ -173,6 +173,38 @@ pub(crate) async fn stop_index(
     Err(ApiError::default())
 }
 
+pub(crate) async fn revert_index(
+    Path((namespace, identifier)): Path<(String, String)>,
+    Extension(tx): Extension<Option<Sender<ServiceRequest>>>,
+    Extension(pool): Extension<IndexerConnectionPool>,
+) -> ApiResult<axum::Json<Value>> {
+    let mut conn = pool.acquire().await?;
+
+    let _ = queries::start_transaction(&mut conn).await?;
```

Comment:
Since `penultimate_index_id_for` is just a `SELECT` I don't think we need a transaction?

Reply:
Maybe I'm misunderstanding the code, but don't we need to fire a `tx.send()` so that `rx.try_recv()` in `service.rs` will receive the value?

Reply:
@rvmelkonian 
- You're correct that we're going to send the results of `penultimate_index_id_for` to the indexer service via `tx.send`
- However, since `penultimate_index_id_for` isn't a mutation query (e.g., an `INSERT` or `DELETE`) then we don't really need a transaction
  - The other use cases in `uses.rs` that use transactions are queries that mutate the state of the database

Reply:
got it, yes we are not mutating here. 
So we need `tx.send` but not this:
```
    let _ = queries::start_transaction(&mut conn).await?;

    if let Err(_e) =
        queries::penultimate_index_id_for(&mut conn, &namespace, &identifier).await
    {
        queries::revert_transaction(&mut conn).await?;
    } else {
        queries::commit_transaction(&mut conn).await?;
    }
    ```